contracts/oracle/PCVOracle.sol

```diff
@@ -0,0 +1,226 @@
+pragma solidity 0.8.13;
+
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import {EnumerableSet} from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+
+import {CoreRefV2} from "../refs/CoreRefV2.sol";
+import {MarketGovernanceOracle} from "./MarketGovernanceOracle.sol";
+
+contract PCVOracle is CoreRefV2 {
+    using SafeCast for *;
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    /// @notice emitted when a new illiquid venue is added
+    event IlliquidVenueAdded(address illiquidVenue);
+
+    /// @notice emitted when a new liquid venue is added
+    event LiquidVenueAdded(address liquidVenue);
+
+    /// @notice emitted when a illiquid venue is removed
+    event IlliquidVenueRemoved(address illiquidVenue);
+
+    /// @notice emitted when a liquid venue is removed
+    event LiquidVenueRemoved(address liquidVenue);
+
+    /// @notice emitted when total liquid venue PCV changes
+    event LiquidVenuePCVUpdated(uint256 oldLiquidity, uint256 newLiquidity);
+
+    /// @notice emitted when total illiquid venue PCV changes
+    event IlliquidVenuePCVUpdated(uint256 oldLiquidity, uint256 newLiquidity);
+
+    /// @notice emitted when market governance oracle is updated
+    event MarketGovernanceOracleUpdated(
+        address oldMgovOracle,
+        address newMgovOracle
+    );
+
+    ///@notice set of whitelisted pcvDeposit addresses for withdrawal
+    EnumerableSet.AddressSet private liquidVenues;
+
+    ///@notice set of whitelisted pcvDeposit addresses for withdrawal
+    EnumerableSet.AddressSet private illiquidVenues;
```

Comment:
mapping of venue address to decimal normalizer, applied on deposit or withdraw from all venues